c-vendor/vendor.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+set -e
+
+# XXX: provide a nice declarative format
+# XXX2: reuse vendor.json and write some python3 code?
+if [ ! -d ./squashfuse ]; then
+    git clone https://github.com/snapcore/squashfuse
+fi
+(cd squashfuse && git checkout 319f6d41a0419465a55d9dcb848d2408b97764f9)
```

Comment:
Why do we need to do this?

Reply:
We are currently "pinning" all vendored dependencies to a git hash. This is the same hash that was used on vendor.json before and I didn't want to do any functional change in this PR. I think we want a followup eventually that moves to a later version of squashfuse and also in a followup we need a version that supports libfuse3 but that is all for the future :) This here is meant to be exactly the same squashfuse as from vendor.json just using a diffrent mechanism to get it.

Reply:
Ah I see, just thought it was curious because 319f6d41a0 is the tip of master but that makes sense. Thanks :slightly_smiling_face: